lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1075,12 +1079,34 @@ ONNXModelLoader::loadConstantOfShape(const ONNX_NAMESPACE::NodeProto &op,
     auto TH = in->getPayload().getHandle<int64_t>();
     llvm::ArrayRef<size_t> outputDims = {(const size_t *)TH.begin(),
                                          (const size_t *)TH.end()};
-    ty = G_.getParent()->uniqueType(ElemKind::FloatTy, outputDims);
+
+    ty = G_.getParent()->uniqueType(T.getType().getElementType(), outputDims);
+    switch (T.getType().getElementType()) {
+    case ElemKind::Int64ITy: {
+      int64_t v = T.getHandle<int64_t>().raw(0);
+      RETURN_ERR_IF_NOT(
+          v == static_cast<int64_t>(static_cast<float>(v)),
+          "This ConstantOfShape implementation may cause losses for value" +
+              std::to_string(v) + " .");
+      SN = G_.createSplat(loadOperatorName(op), ty, v);
+      break;
+    }
+    case ElemKind::Int32ITy: {
+      int32_t v = T.getHandle<int32_t>().raw(0);
+      RETURN_ERR_IF_NOT(
+          v == static_cast<int32_t>(static_cast<float>(v)),
```

Comment:
could you please add a test when this fails as well.

Reply:
added